src/profile/profile.ts

```diff
@@ -478,6 +491,22 @@ export class Profile extends Logger {
     );
   }
 
+
+  /**
+   * Gets the identity list.
+   */
+  public async getIdentityAccessList(): Promise<any> {
+    await this.getAddressBook();
+    const { keys } = this.trees[this.treeLabels.addressBook];
+    const filteredIdentities = Object.keys(keys)
+      .filter((key) => keys[key].identityAccess);
+    const result = {};
+    for (const identity of filteredIdentities) {
+      result[identity] = keys[identity];
+    }
+    return result;
```

Comment:
Wouldn't something like this suffice? (Don't commit it without checking :D)
```suggestion
    return filteredIdentities.map((identity) => ({ [identity]: keys[identity] }))
```